atomix/cluster/src/main/java/io/atomix/raft/impl/RaftContext.java

```diff
@@ -512,6 +512,24 @@ public CompletableFuture<Void> compact() {
     return future;
   }
 
+  /**
+   * Ensures everything written to the log until this point, is flushed to disk.
+   *
+   * @return a future to be completed once the log is flushed to disk
+   */
+  public CompletableFuture<Void> flushLog() {
+    final CompletableFuture<Void> future = new CompletableFuture<>();
+    threadContext.execute(
+        () -> {
+          // Only flush if explicit flush is disabled
+          if (!raftLog.shouldFlushExplicitly()) {
+            raftLog.flush();
+          }
```

Comment:
:thinking: Wondering whether this is really necessary? Do you think this is an issue if we do it anyway without checking?

Reply:
:x: If we want this, then we should also document it as part of the method documentation.

Reply:
> Wondering whether this is really necessary? Do you think this is an issue if we do it anyway without checking?

It is more of an optimization. Since this is running on raft thread, we try to remove an unnecessary flush that can cause an i/o operation. If flush is not disabled, then the log are flushed before commit. So there is no need to flush again here.

Reply:
I will update the docs.